### DISCUSSION TITLE
Get Durable Function Status and Metadata in Middleware - i.e. InstanceId?

### DISCUSSION THREAD

**Discussion by dfaivre** [Azure/azure-functions-durable-extension#2395] [Q&A]
I'm trying to capture some durable function status to link back to our business objects using middleware. Is there a way to get any of the Durable Functions metadata, like the `InstanceId` or the `ParentOrchestrationInstance` for an Orchestration within the middleware?

**Comment by cgillum** [comment]
Just to clarify, are you referring to middleware that runs in the .NET Isolated worker process?

**Reply by LockTar** [reply]
@cgillum is there a solution for this? I need the instanceId in the middleware.
I'm running on the latest version of Durable Functions Isolated.

**Comment by dfaivre** [comment]
Yes - sorry should have included that
